File: split/newparamglue.pas
function newparamglue(n: smallnumber): halfword;
var
  p: halfword;
  q: halfword;
begin
  p := getnode(2);
  mem[p].hh.b0 := 10;
  mem[p].hh.b1 := n + 1;
  mem[p + 1].hh.rh := 0;
  q :={224:}
eqtb[2882 + n].hh.rh{:224}
;
  mem[p + 1].hh.lh := q;
  mem[q].hh.rh := mem[q].hh.rh + 1;
  newparamglue := p;
end;

File: cpp/newparamglue.cpp
#include "newparamglue.h"
#include "getnode.h"

halfword newparamglue(smallnumber n)
{
	auto p = getnode(2);
	type(p) = glue_node;
	subtype(p) = n+1;
	link(p+1) = 0;
	auto q = glue_par(n);
	info(p+1) = q;
	link(q)++;
	return p;
}
